### ISSUE TITLE
arXiv'2020 | Class-incremental learning: survey and performance evaluation on image classification

### ISSUE THREAD

**Issue by swagshaw** [swagshaw/ResearchReading#1]
https://arxiv.org/pdf/2010.15277.pdf

**Comment by swagshaw** [comment]
![image](https://user-images.githubusercontent.com/48468760/152334201-9d7e76ab-1e50-4f0e-8556-c9828c2cb24f.png)
Relation between class-IL methods. We distinguish three main categories: exemplar-free regularization (green), rehearsal (blue), and rehearsal with explicit bias-correction mechanisms (orange).

**Comment by swagshaw** [comment]
![image](https://user-images.githubusercontent.com/48468760/152627252-1301e4d1-24e8-49f2-8811-928167ca195b.png)
In incremental learning, disjoint tasks are learned sequentially. Task-IL has access to the task-ID during evaluation, while the more challenging setting of class-IL does not.

**Comment by swagshaw** [comment]
### Background
For future learning systems incremental learning is desirable because it allows for: efficient resource usage by eliminating the need to retrain from scratch at the arrival of new data; reduced memory usage by preventing or limiting the amount of data required to be stored – also important when privacy limitations are imposed; and learning that more closely resembles human learning.

**Comment by swagshaw** [comment]
### Challenge
The main challenge for incremental learning is catastrophic forgetting, which refers to the precipitous drop in performance on previously learned tasks after learning a new one.

- **Weight drift**: While learning new tasks, the network weights relevant to old tasks are updated to minimize a loss on the new task.
- **Activation drift**: Closely related to weight drift, changing weights result in changes to activations, and consequently in changes to the network output.
- **Inter-task confusion**: in class-IL the objective is to discriminate all classes from all tasks. However, since classes are never jointly trained the network weights cannot optimally discriminate all classes
- **Task-recency bias**: Separately learned tasks might have incomparable classifier outputs. Typically, the most dominant task bias is towards more recent task classes.

**Comment by swagshaw** [comment]
### Exist Solutions

1. **regularization-based** solutions that aim to minimize the impact of learning new tasks on the weights that are important for previous tasks
2. **exemplar-based** solutions that store a limited set of exemplars to prevent forgetting of previous tasks
3. solutions that directly address the problem of **task-recency bias**, a phenomenon occurring in class-IL methods that refers to the bias towards recently-learned tasks.

### regularization-based

1. Weight regularization: Elastic Weight Consolidation(EWC), Path Integral(PathInt), Aware Synapses (MAS), Riemanian Walk (RWalk)
2. Data regularization: Learning without Forgetting (LwF), less-forgetting learning (LFL)
3. Recent developments in regularization: Learning without Memorizing (LwM), Deep Model Consolidation (DMC), Global Distillation (GD), less-forget constraint

### Rehearsal approaches

Incremental Classifier and Representation Learning (iCaRL), End-to-End Incremental Learning (EEIL), Global Distillation (GD), and so on

### Bias-correction approaches
Bias Correction (BiC), Learning a Unified Classifier Incrementally via Rebalancing (LUCIR)

**Weight regularization methods, applied frequently in the task-IL setting, are significantly less used for class-IL.**

### Other approaches
#### Mask-based methods.
Piggyback, PackNet, PathNet
#### Dynamic architectures.
PNN, ACL, Random Path Selection (RPS)
#### Pseudo-rehearsal methods.
DGR, Lifelong GAN, feature replay methods, Incremental learning using conditional adversarial networks, Fearnet: Brain inspired model for incremental learning

**Comment by swagshaw** [comment]
### Emerging Trends in CLASS-IL
1. Exemplar learning. Recently, an exciting new direction has emerged that parametrizes exemplars and optimizes them to prevent forgetting [41], [110].
2. Feature rehearsal. Pseudo-rehearsal is a good alternative to storing exemplars [29], [30], [47].
3. Self- and unsupervised incremental learning. Being able to incrementally learn representations from an unsupervised data stream is a desirable feature in any learning system.[114] [115]
4. Beyond cross-entropy loss. Less forgetting has been reported by replacing the cross-entropy loss with a metric learning loss [116] or by using an energy-based method [117].
5. Meta-learning. Meta-learning aims to learn new tasks leveraging information accrued while solving related tasks.[71][119]
6. Task-free settings. A more realistic scenario is one where there are no clear task boundaries and the distribution over classes changes gradually. [117], [120], [121].

[41] Y. Liu, A.-A. Liu, Y. Su, B. Schiele, and Q. Sun, “Mnemonics training:
Multi-class incremental learning without forgetting,” in Proc. IEEE
Conf. Comput. Vis. Pattern Recognit., 2020.
[110] A. Chaudhry, A. Gordo, P. K. Dokania, P. Torr, and D. Lopez-Paz,
“Using hindsight to anchor past knowledge in continual learning,” Proc.
AAAI Conf. Artif. Intell., 2021.


[29] H. Shin, J. K. Lee, J. Kim, and J. Kim, “Continual learning with deep
generative replay,” in Proc. Adv. Neural Inf. Process. Syst., 2017.
[30] C. Wu, L. Herranz, X. Liu, Y. Wang, J. van de Weijer, and B. Raducanu,
“Memory replay GANs: learning to generate images from new
categories without forgetting,” in Proc. Adv. Neural Inf. Process. Syst.,
2018.
[47] O. Ostapenko, M. Puscas, T. Klein, P. J¨ahnichen, and M. Nabi, “Learning
to remember: A synaptic plasticity driven framework for continual
learning,” in Proc. IEEE Conf. Comput. Vis. Pattern Recognit., 2019.

[114] D. Rao, F. Visin, A. Rusu, R. Pascanu, Y. W. Teh, and R. Hadsell,
“Continual unsupervised representation learning,” in Proc. Adv. Neural
Inf. Process. Syst., 2019.
[115] L. Jing and Y. Tian, “Self-supervised visual feature learning with deep
neural networks: A survey,” IEEE Trans. Pattern Anal. Mach. Intell.,
2020.
[116] L. Yu, B. Twardowski, X. Liu, L. Herranz, K. Wang, Y. Cheng, S. Jui,
and J. v. d. Weijer, “Semantic drift compensation for class-incremental
learning,” in Proc. IEEE Conf. Comput. Vis. Pattern Recognit., 2020.
[117] S. Li, Y. Du, G. M. van de Ven, A. Torralba, and I. Mordatch, “Energybased
models for continual learning,” arXiv, 2020.

[71] M. Riemer, I. Cases, R. Ajemian, M. Liu, I. Rish, Y. Tu, and
G. Tesauro, “Learning to learn without forgetting by maximizing transfer
and minimizing interference,” in Proc. Int. Conf. Learn. Repres.,
2019.
[119] K. Javed and M. White, “Meta-learning representations for continual
learning,” in Proc. Adv. Neural Inf. Process. Syst., 2019.

[120] R. Aljundi, K. Kelchtermans, and T. Tuytelaars, “Task-free continual
learning,” in Proc. IEEE Conf. Comput. Vis. Pattern Recognit., 2019.
[121] J. Rajasegaran, S. Khan, M. Hayat, F. S. Khan, and M. Shah, “iTAML:
An incremental task-agnostic meta-learning approach,” in Proc. IEEE
Conf. Comput. Vis. Pattern Recognit., 2020.
